mmdet/models/layers/transformer/dino_transformer.py

```diff
@@ -0,0 +1,113 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+import math
+
+import torch
+from torch import nn
+
+from .deformable_detr_transformer import DeformableDetrTransformerDecoder
+from .utils import MLP, inverse_sigmoid
+
+
+class DinoTransformerDecoder(DeformableDetrTransformerDecoder):
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self._init_layers()
+
+    def _init_layers(self):
+        super()._init_layers()
+        self.ref_point_head = MLP(self.embed_dims * 2, self.embed_dims,
+                                  self.embed_dims, 2)
+        self.norm = nn.LayerNorm(self.embed_dims)  # TODO: refine this
+
+    @staticmethod
+    def gen_sineembed_for_position(pos_tensor):
```

Comment:
pos_tensor seems a points or bbox?
if yes, pos_tensor seems confusing

Reply:
@jshilong  Yes, it is also used in Conditional DETR and DAB DETR, what about name it with `positions`?